gpt_engineer/ai.py

```diff
@@ -34,12 +54,18 @@ def next(self, messages: list[dict[str, str]], prompt=None):
             messages += [{"role": "user", "content": prompt}]
 
         logger.debug(f"Creating a new chat completion: {messages}")
-        response = openai.ChatCompletion.create(
-            messages=messages,
-            stream=True,
-            model=self.model,
-            temperature=self.temperature,
-        )
+
+        shared_params = {
+            "messages": messages,
+            "stream": True,
+            "temperature": self.temperature,
+        }
+        if not self.is_azure:
+            # use OpenAI's models directly
+            response = openai.ChatCompletion.create(model=self.model, **shared_params)
+        else:
+            # use Azure OpenAI
+            response = openai.ChatCompletion.create(engine=self.model, **shared_params)
```

Comment:
I realize that if we want to support this we should in fact have a separate object:

The LLM model.

I'd be happy to merge if you find a good way to "abstract away" the LLM object!